lib/src/main/kotlin/com/swmansion/starknet/data/serializers/JsonRpcTransactionPayloadPolymorphicSerializer.kt

```diff
@@ -0,0 +1,16 @@
+package com.swmansion.starknet.data.serializers
+
+import com.swmansion.starknet.data.types.transactions.*
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.json.*
+
+internal object JsonRpcTransactionPayloadPolymorphicSerializer :
+    JsonContentPolymorphicSerializer<TransactionPayload>(TransactionPayload::class) {
+    override fun selectDeserializer(element: JsonElement): DeserializationStrategy<out TransactionPayload> =
+        when (element.jsonObject["type"]?.jsonPrimitive?.content) {
+            "INVOKE" -> InvokeTransactionPayload.serializer()
+            "DECLARE" -> DeclareTransactionPayload.serializer()
+            "DEPLOY_ACCOUNT" -> DeployAccountTransactionPayload.serializer()
+            else -> throw IllegalArgumentException("Invalid transaction type '${element.jsonObject["type"]?.jsonPrimitive?.content}'")
```

Comment:
If either of `?` calls is not successful, I believe the resulting string will be `"Invalid transaction type 'null'`. Do you think a default value other than that would be useful? It could be provided inside the ${} block using `:? "default"`

Reply:
I figured, that if that's the case we should be fine with just 'null', since it actually will be null; pointing us to the fact that the response is invalid; perhaps the message should be changed to reflect that somehow?

Reply:
Maybe creating handling for cases that are not very common like this is not entirely needed. I though that maybe we could have a separate error messages for a) type not being valid b) type missing entirely. But if you think it's not necessary, we can leave it as it is.